ui/browser/actions/main-actions.js

```diff
@@ -46,3 +46,7 @@ export function setPageOrder(pageOrder) {
 export function setPageAreaVisibility(visible) {
   return { type: types.SET_PAGE_AREA_VISIBILITY, visible };
 }
+
+export function clickTab(pageIndex, button) {
+  return (button === 1) ? closeTab(pageIndex) : setCurrentTab(pageIndex);
```

Comment:
Yeah, I would move this conditional logic back into the component instead of adding a new action